nni/retiarii/oneshot/pytorch/base_lightning.py

```diff
@@ -0,0 +1,169 @@
+        custom_replace_dict : dict{ type : callable }, default = None
+            The custom xxxChoice replace method. Keys should be xxxChoice type and values should 
+            return an nn.module. This custom replace dict will override the default replace
+            dict of each nas method.
+        '''
+        super().__init__()
+        assert isinstance(base_model, pl.LightningModule)
+        self.model = base_model
+
+        # replace xxxChoice with respect to nas alg
+        # replaced modules are stored in self.nas_modules
+        self.nas_modules = []
+        choice_replace_dict = self.default_replace_dict
+        if custom_replace_dict is not None:
+            for k,v in custom_replace_dict.items():
+                assert isinstance(v, nn.Module)
+                choice_replace_dict[k] = v
+        _replace_module_with_type(self.model, choice_replace_dict, self.nas_modules)
+    
+    def __getattr__(self, name):
+        ''' 
+        Redirect lightning hooks. 
+        Only hooks in the list below will be redirect to user-deifend ones.
+        The list is not customizable now. But if you need to redirect other user-defined
+        hooks in your nas method, you can override it and call user-defined methods manually.
+
+        Warnings
+        ----------
+        Validation-related hook are bypassed as default, since architecture selection
+        is exactly what nas aims to do.
+        '''
+        if name in [
+            'on_train_end', 'on_fit_start', 'on_fit_end',
+            'on_train_batch_start', 'on_train_batch_end', 
+            'on_epoch_start', 'on_epoch_end', 
+            'on_train_epoch_start', 'on_train_epoch_end', 
+            'on_before_backward', 'on_after_backward',
+            'configure_gradient_clipping'
+        ]:
+            return getattr(self.__dict__['_modules']['model'], name)    
+        return super().__getattr__(name)
+        
+
+    def forward(self, x):
+        return self.model(x)
+
+    def training_step(self, batch, batch_idx):
+        return self.model.training_step(batch, batch_idx)
+    
+    def validation_step(self, batch, batch_idx):
+        warning('Validation is skipped by the nas method you chose.')
+
+    def configure_optimizers(self):
+        '''
+        Combine architecture optimizers and user's model optimizers.
+        Overwrite configure_architecture_optimizers if architecture optimizers
+        are needed in your NAS algorithm.
+        '''
+        arc_optimizers = self.configure_architecture_optimizers()
+        if arc_optimizers is None:
+            return self.model.configure_optimizers()
+        
+        if isinstance(arc_optimizers, optim.Optimizer):
+            arc_optimizers = [arc_optimizers]
+        self.arc_optim_count = len(arc_optimizers)
+
+        w_optimizers = self.model.configure_optimizers()
+        if isinstance(w_optimizers, optim.Optimizer):
```

Comment:
There are six different options to `configure_optimizers`, only two are handled here I think.

https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers